### ISSUE TITLE
Reflection skips properties of type Dictionary<(Tuple with nullable type), T>

### ISSUE THREAD

**Issue by JuliusSweetland** [protobuf-net/protobuf-net.Grpc#288]
I have a property as below. When I enable reflection and attempt to dump a proto file from the running service using the command below the proto is generated, but is missing this property. Experimenting it works if I replace the tuple with a struct of my own (e.g. public struct LongAndNullableDouble), or if I make the double non-nullable then the property is included in the dumped proto.

Command:
```

```
[ProtoMember(3)]
public Dictionary<(long, double?), MyType>? MyProperty { get; set; }
```

**Comment by mgravell** [comment]
Nope; I saw it initially - looks very curious, but I have not yet had chance to investigate. I'm intrigued - I just don't have infinite time :)

**Comment by JuliusSweetland** [comment]
@mgravell Completely understand. I have an open source project as well and appreciate the pain.

**Comment by JuliusSweetland** [comment]
Is anyone able to give a bit of guidance on this one? Thanks.
